File: Examples/VisualBasic/Hyperlinks/AddHyperlink.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing
Namespace Hyperlinks
    Public Class AddHyperlink
        Public Shared Sub Run()
            ' ExStart:AddHyperlink
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tasks()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New Aspose.Note.Page(doc)
            ' Initialize Title class object
            Dim title As New Title(doc)

            Dim defaultTextStyle As New TextStyle() With {
                 .FontName = "Arial",
                 .FontSize = 10,
                 .FontColor = SystemColors.WindowText
            }
            Dim titleText As New RichText(doc) With {
                 .Text = "Title!",
                 .DefaultStyle = defaultTextStyle
            }
            Dim outline As New Outline(doc) With {
                 .MaxWidth = 200,
                 .MaxHeight = 200,
                 .VerticalOffset = 100,
                 .HorizontalOffset = 100
            }
            Dim outlineElem As New OutlineElement(doc)
            'this style will be applied to 0-7 characters.
            Dim textStyleRed As New TextStyle() With {
                 .FontColor = Color.Red,
                 .FontName = "Arial",
                 .FontSize = 10,
                 .RunIndex = 8
            }
            'this style will be applied to 8-16 characters.
            Dim textStyleHyperlink As New TextStyle() With {
                 .RunIndex = 17,
                 .IsHyperlink = True,
                 .HyperlinkAddress = "www.google.com"
            }
            Dim text As New RichText(doc) With {
                 .Text = "This is hyperlink. This text is not a hyperlink.",
                 .DefaultStyle = defaultTextStyle
            }

            title.TitleText = titleText
            page.Title = title
            outlineElem.AppendChild(Text)
            ' Add outline elements
            Outline.AppendChild(outlineElem)
            ' Add Outline node
            page.AppendChild(Outline)
            ' Add Page node
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("AddHyperlink_out_.one")
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:AddHyperlink
            Console.WriteLine(Convert.ToString(vbLf & "Hyperlink added successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/Hyperlinks/AddHyperlink.cs
// -----------------------------------------------------------------------
//  <copyright file="AddHyperlink.cs" company="Aspose Pty Ltd">
//    Copyright (c) 2002-2022 Aspose Pty Ltd. All Rights Reserved.
//  </copyright>
// -----------------------------------------------------------------------

namespace Aspose.Note.Examples.CSharp.Hyperlinks
{
    using System;
    using System.Drawing;

    public class AddHyperlink
    {
        public static void Run()
        {
            // ExStart:AddHyperlink
            // ExFor:Document
            // ExFor:RichText
            // ExFor:RichText.ParagraphStyle
            // ExFor:RichText.Text
            // ExFor:RichText.Append(System.String)
            // ExFor:RichText.Append(System.String,TextStyle)
            // ExFor:TextStyle
            // ExFor:Style.FontColor
            // ExFor:Style.FontName
            // ExFor:Style.FontSize
            // ExFor:TextStyle.IsHyperlink
            // ExFor:TextStyle.HyperlinkAddress
            // ExSummary:Shows how to bind a hyperlink to a text.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tasks();

            // Create an object of the Document class
            Document doc = new Document();


            RichText titleText = new RichText() { ParagraphStyle = ParagraphStyle.Default }.Append("Title!");

            Outline outline = new Outline()
                                  {
                                      MaxWidth = 200,
                                      MaxHeight = 200,
                                      VerticalOffset = 100,
                                      HorizontalOffset = 100
                                  };

            TextStyle textStyleRed = new TextStyle
                                         {
                                             FontColor = Color.Red,
                                             FontName = "Arial",
                                             FontSize = 10,
                                         };

            TextStyle textStyleHyperlink = new TextStyle
                                               {
                                                   IsHyperlink = true,
                                                   HyperlinkAddress = "www.google.com"
                                               };

            RichText text = new RichText() { ParagraphStyle = ParagraphStyle.Default }
                                .Append("This is ", textStyleRed)
                                .Append("hyperlink", textStyleHyperlink)
                                .Append(". This text is not a hyperlink.", TextStyle.Default);

            OutlineElement outlineElem = new OutlineElement();
            outlineElem.AppendChildLast(text);
            
            // Add outline elements
            outline.AppendChildLast(outlineElem);

            // Initialize Title class object
            Title title = new Title() { TitleText = titleText };

            // Initialize Page class object
            Page page = new Note.Page() { Title = title };

            // Add Outline node
            page.AppendChildLast(outline);

            // Add Page node
            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "AddHyperlink_out.one";
            doc.Save(dataDir);
            
            // ExEnd:AddHyperlink
            
            Console.WriteLine("\nHyperlink added successfully.\nFile saved at " + dataDir);
        }
    }
}